haystack/components/embedders/azure_document_embedder.py

```diff
@@ -0,0 +1,182 @@
+import os
+from typing import List, Optional, Dict, Any, Tuple
+
+from openai.lib.azure import AzureADTokenProvider, AzureOpenAI
+from tqdm import tqdm
+
+from haystack import component, Document, default_to_dict
+
+
+@component
+class AzureOpenAIDocumentEmbedder:
+    """
+    A component for computing Document embeddings using OpenAI models.
+    The embedding of each Document is stored in the `embedding` field of the Document.
+
+    Usage example:
+    ```python
+    from haystack import Document
+    from haystack.components.embedders import AzureOpenAIDocumentEmbedder
+
+    doc = Document(content="I love pizza!")
+
+    document_embedder = AzureOpenAIDocumentEmbedder()
```

Comment:
Just a doubt. Does this example work without specifying any parameter?

Reply:
Yes, if you specify AZURE_OPENAI_API_KEY and AZURE_OPENAI_ENDPOINT env vars. That's why I kept them optional (the respecting init params). This is the recommended approach in Azure [docs](https://learn.microsoft.com/en-us/azure/ai-services/openai/tutorials/embeddings?tabs=python%2Ccommand-line&pivots=programming-language-python)